test/pylib/report_plugin.py

```diff
@@ -17,7 +17,10 @@ class ReportPlugin:
 
     # Pytest hook to modify test name to include mode and run_id
     def pytest_configure(self, config):
-        self.build_mode = config.getoption('mode')
+        # getting build_mode is needed for the cases when there will be no mode provided
+        self.build_mode = config.getoption("modes")
+        if self.build_mode:
+            self.build_mode = self.build_mode[0]
```

Comment:
This is needed for the old way of executing tests with test.py. All tests there work only with one mode, so they will continue to work like that. Boost/unit test can utilize modes as list, and they are not using this plugin. So they will be working in a little bit different way. Later on, it will be synchronized among all tests.